nomad/structs/structs.go

```diff
@@ -7253,6 +7265,10 @@ type Allocation struct {
 	ModifyTime int64
 }
 
+// AllocationDiff is a type alias for Allocation used to indicate that a diff is
+// and not the entire allocation
+type AllocationDiff = Allocation
```

Comment:
Using a named type would have been a better idea as an AllocationDiff can't be replaced by Allocation on that case. The downside was that AllocationDiff wouldn't have the same methods defined on the struct, that Allocation does, which probably means rewriting code.

Reply:
Do we want the same methods defined? Allocation has a lot of methods that will not work or worse: appear to work but not have the intended result.

If there's a small number of methods that need to be shared AllocationDiff I think a named type can be converted easily:

```go
func (ad *AllocationDiff) CopySkipJob() *Allocation {
    return Allocation(ad).CopySkipJob()
}
```

If that's more trouble than it's worth let's remove this type alias, use Allocation directly, and carefully comment the fields that have partial Allocs.